sdk/src/types/block/output/feature/mod.rs

```diff
@@ -143,6 +153,21 @@ impl Feature {
             panic!("as_tag called on a non-tag feature");
         }
     }
+
+    /// Checks whether the feature is a [`StakingFeature`].
+    pub fn is_staking(&self) -> bool {
+        matches!(self, Self::Staking(_))
+    }
+
+    /// Gets the feature as an actual [`StakingFeature`].
+    /// PANIC: do not call on a non-staking feature.
+    pub fn as_staking(&self) -> &StakingFeature {
+        if let Self::Staking(feature) = self {
+            feature
+        } else {
+            panic!("as_staking called on a non-staking feature");
+        }
+    }
```

Comment:
out of interest ... why do we panic for those `as_` methods instead of returning a `Result` or even just an `Option`?

Reply:
Convenience

Reply:
ah, okay. Is that something we should reconsider doing because for the user of the library it may be more convenient if it's possible to recover.

Reply:
I wouldn't, there is a comment saying it panics and there is an `is_` method to check before if they really want. There are plenty of situations where we know for sure the type and it's just unnecessary to have to unwrap something.